error/s2n_errno.h

```diff
@@ -227,7 +228,7 @@ extern __thread const char *s2n_debug_str;
 #define STRING__LINE__ STRING_(__LINE__)
 
 #define _S2N_DEBUG_LINE     "Error encountered in " __FILE__ " line " STRING__LINE__
-#define _S2N_ERROR( x )     do { s2n_debug_str = _S2N_DEBUG_LINE; s2n_errno = ( x ); } while (0)
+#define _S2N_ERROR( x )     do { s2n_debug_str = _S2N_DEBUG_LINE; s2n_errno = ( x ); s2n_calculate_stacktrace(); } while (0)
```

Comment:
hmm looks like calculating and printing the stacktrace are done in separate steps, wonder if there's a prettier way to doing this..

Reply:
I can add a function that does both.  The issue is that I don't want to print stacktraces on every error, since unit-tests often have cases where errors are expected.

Reply:
I see... in my development env I use a macro called S2N_STACKTRACE that I insert when I need it